crates/binjs_io/src/io/mod.rs

```diff
@@ -239,33 +221,24 @@ pub trait TokenReader: FileStructurePrinter where Self::ListGuard: Guard,
     fn offset_at(&mut self, _path: &Path) -> Result<u32, TokenReaderError>;
 
     /// Start reading a list.
-    ///
-    /// Returns a guard for that list and the number of elements
-    /// in the list. Once the list is read entirely, callers MUST
-    /// call `guard.done()` to ensure that the list was properly
-    /// read (in particular that all bytes were consumed). In most
-    /// implementations, failure to do so will raise an assertion.
-    fn list_at(&mut self, _path: &Path) -> Result<(u32, Self::ListGuard), TokenReaderError>;
+    fn enter_list_at(&mut self, _path: &Path) -> Result<u32, TokenReaderError>;
+
+    /// Finish reading a list.
+    fn leave_list_at(&mut self, _path: &Path) -> Result<(), TokenReaderError>;
```

Comment:
On the writer side, it's called `enter_`/`exit_`, could you keep that naming convention?